tools/sr-snapshots/Sources/SRSnapshotsCore/Commands.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Git
+import Files
+import Foundation
+import ArgumentParser
+
+/// Git clone ssh for snapshots repo.
+private let ssh = "git@github.com:DataDog/dd-mobile-session-replay-snapshots.git"
```

Comment:
question

how would cloning via ssh work in CI? Are credentials baked in the CI environment?

Reply:
This is not the nicest solution, but the actual detail is implemented in `BasicGitClient` that relies on invoking shell commands directly. One of the commands uses [gh](https://cli.github.com/) which is an official GitHub CLI client. The [`gh repo clone`](https://cli.github.com/manual/gh_repo_clone) is capable of cloning any internal repo [using ENV authentication](https://cli.github.com/manual/gh_auth_login) (`GH_TOKEN`).

This setup already exists in our repo as I explained in https://github.com/DataDog/dd-sdk-ios/pull/1299#discussion_r1214541130.

Historically, we found `gh repo clone` easier than provisioning CI with SSH keys.